pkg/scheduler/util/utils_test.go

```diff
@@ -173,3 +174,72 @@ func TestRemoveNominatedNodeName(t *testing.T) {
 		})
 	}
 }
+
+func TestPatchPodStatus(t *testing.T) {
+	tests := []struct {
+		name           string
+		podName        string
+		secrets        []v1.LocalObjectReference
+		statusToUpdate v1.PodStatus
+	}{
+		{
+			name:    "Should update pod conditions successfully",
+			podName: "pod2",
+			secrets: []v1.LocalObjectReference{{Name: "foo"}},
```

Comment:
This makes the test table specific to this bug. Better to accept a Pod.

Reply:
yeah I realized what you meant by having the secret as part of the test table. updated now :)